### ISSUE TITLE
Replace smileys with emoji everywhere

### ISSUE THREAD

**Issue by koke** [wordpress-mobile/WordPress-iOS#1188]
We're already using `stringByReplacingHTMLEmoticonsWithEmoji` for notifications. Make sure the reader and other parts that display content use it as well

**Comment by jleandroperez** [comment]
Closing this one since Notifications actually rely on `Ranges` (analog to NSRange) to deal with metadata. Tampering with the string content would definitely break in multiple ways.

Please, feel free of reopening if needed.

**Comment by koke** [comment]
I think that can be done safely at the view layer, when setting up the `NoteBlockCommentTableViewCell`. I don't mind leaving this closed if we're not going to work on it, but I think it's doable. For comparison:

Comment on the web

![screen shot 2016-11-09 at 09 55 19](https://cloud.githubusercontent.com/assets/8739/20133162/a613a7e0-a666-11e6-8fc2-1afcd98477b0.png)

Comment on calypso

![screen shot 2016-11-09 at 09 55 27](https://cloud.githubusercontent.com/assets/8739/20133194/c59c978e-a666-11e6-93b6-a1827b647648.png)

Comment on the app

![simulator screen shot 9 nov 2016 09 55 35](https://cloud.githubusercontent.com/assets/8739/20133204/cb721a80-a666-11e6-90b9-a7ed72668495.png)

**Comment by jleandroperez** [comment]
You're right @koke. I've overlooked this. As you pointed out, we can do this in `NotificationBlock+Interface`, which is the precise spot where the `NSRange` metadata is applied.

Should be as simple as calling `replaceCharactersInRange`, and accounting for any change in length.

@koke do you know if we've got, somewhere in Calypso, an "Emoji Conversion Table" or so?

Thanks!!

**Comment by koke** [comment]
It's not handled by calypso but by the backend. Smilies are a core thing, with a bunch of extra wp.com smilies on top. The odd thing is that those aren't currently replaced in notifications.

For comments and reader content, the existing code doesn't look for the smiley text itself but it replaces the image embed with an emoji. However, for notifications the text is present instead of the image. Which makes me think we might need to fix it on the API side.

**Comment by yaelirub** [comment]
Still an issue in Comments. Weirdly some emojis are displayed correctly :
![Screen Shot 2020-06-09 at 1 13 39 PM](https://user-images.githubusercontent.com/1335657/84195254-1afcbd00-aa53-11ea-87ce-ce8cf8cc3700.png)

**Comment by aerych** [comment]
~Closing. Seems resolved. I tested with a new comment with the text:~
```
:)
:smile: 
😀
```
~And all three formats were shown as emoji in notifications.
If we find new cases where emoji is not rendered we can revisit.~

Realizing this issue isn't about notifications, but other parts of the app.  Removing the notifications label and replacing it with the general label.

**Comment by dangermattic** [comment]
Thanks for reporting! 👍
